corebuild/integration/ILLink.Tasks/ILLink.CrossGen.targets

```diff
@@ -0,0 +1,231 @@
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+
+  <PropertyGroup>
+    <CrossGenDuringPublish Condition=" '$(CrossGenDuringPublish)' == '' And '$(RuntimeIdentifier)' != '' ">true</CrossGenDuringPublish>
+  </PropertyGroup>
+
+  <PropertyGroup>
+    <IntermediateOptimizedDirName Condition=" '$(IntermediateOptimizedDirName)' == '' ">optimized</IntermediateOptimizedDirName>
+    <IntermediateOptimizedDir Condition=" '$(IntermediateOptimizedDir)' == '' ">$(IntermediateOutputPath)$(IntermediateOptimizedDirName)</IntermediateOptimizedDir>
+    <IntermediateCrossGenDirName Condition=" '$(IntermediateCrossGenDirName)' == '' ">crossgen</IntermediateCrossGenDirName>
+    <IntermediateCrossGenDir Condition=" '$(IntermediateCrossGenDir)' == '' ">$(IntermediateOutputPath)$(IntermediateCrossGenDirName)</IntermediateCrossGenDir>
+  </PropertyGroup>
+
+  <PropertyGroup>
+    <ComputeCrossGenFilesToPublishAfter Condition=" '$(LinkDuringPublish)' == 'true' ">ComputeLinkedFilesToPublish</ComputeCrossGenFilesToPublishAfter>
+    <ComputeCrossGenFilesToPublishBefore Condition=" '$(LinkDuringPublish)' != 'true' ">ComputeFilesToPublish</ComputeCrossGenFilesToPublishBefore>
+  </PropertyGroup>
+
+  <!-- Rewrite inputs to ComputeFilesToPublish, so that crossgen'd
+       files get published instead of the pre-crossgen assemblies. -->
+  <Target Name="ComputeCrossGenFilesToPublish"
+          BeforeTargets="$(ComputeCrossGenFilesToPublishBefore)"
+          AfterTargets="$(ComputeCrossGenFilesToPublishAfter)"
+          Condition=" '$(CrossGenDuringPublish)' == 'true' "
+          DependsOnTargets="_CrossGenForPublish">
+
+    <!-- Rewrite ResolvedAssembliesToPublish, preserving metadata. -->
+    <ItemGroup>
+      <__CrossGenResolvedAssembliesToPublish Include="@(ResolvedAssembliesToPublish->'$(IntermediateOptimizedDir)/%(Filename)%(Extension)')" />
+      <_CrossGenResolvedAssembliesToPublish Include="@(__CrossGenResolvedAssembliesToPublish)" Condition="Exists('%(Identity)')" />
+      <!-- FilesToCrossGen doesn't contain System.Private.CoreLib, so
+           it will still be published. -->
+      <ResolvedAssembliesToPublish Remove="@(FilesToCrossGen)" />
+      <ResolvedAssembliesToPublish Include="@(_CrossGenResolvedAssembliesToPublish)" />
+    </ItemGroup>
+
+    <!-- Rewrite IntermediateAssembly, preserving metadata. -->
+    <ItemGroup>
+      <__CrossGenIntermediateAssembly Include="@(IntermediateAssembly->'$(IntermediateOptimizedDir)/%(Filename)%(Extension)')" />
+      <_CrossGenIntermediateAssembly Include="@(__CrossGenIntermediateAssembly)" Condition="Exists('%(Identity)')" />
```

Comment:
Can you make the two items names above differ in something other than the number of leading underscores? Do you need the two items?